tools/src/bin/dependency_check.rs

```diff
@@ -0,0 +1,297 @@
+        Some(path) => path.to_path_buf(),
+        None => bail!("Failed to get current executable directory."),
+    };
+
+    let lib_exists = fs::exists(executable_dir.join(FFMPEG_LIB_DIR).join(".ready"))
+        .with_context(|| "Failed to check if local ffmpeg lib directory exists.")?;
+    if lib_exists {
+        return Ok(());
+    } else {
+        let malformed_lib_exists = fs::exists(executable_dir.join(FFMPEG_LIB_DIR))?;
+        if malformed_lib_exists {
+            fs::remove_dir_all(executable_dir.join(FFMPEG_LIB_DIR))?;
+        }
+        let gz_archive_exists = fs::exists(executable_dir.join(FFMPEG_GZIP_ARCHIVE_NAME))?;
+        if gz_archive_exists {
+            fs::remove_file(executable_dir.join(FFMPEG_GZIP_ARCHIVE_NAME))?;
+        }
+    }
+
+    let ffmpeg_installed = check_ffmpeg();
+    match ffmpeg_installed {
+        Ok(true) => return Ok(()),
+        Ok(false) => {
+            info!("Downloading dependencies...");
+            prepare_dependencies(&executable_dir)
+                .with_context(|| "Failed to fetch dependencies.")?;
+        }
+        Err(error) => {
+            error!(%error);
+            info!("Downloading dependencies...");
+            prepare_dependencies(&executable_dir)
+                .with_context(|| "Failed to fetch dependencies.")?;
+        }
+    }
+    cleanup(&executable_dir);
+
+    fs::write(executable_dir.join(FFMPEG_LIB_DIR).join(".ready"), "")?;
+    Ok(())
+}
+
+fn cleanup(executable_dir: &Path) {
+    let gz_archive = executable_dir.join(FFMPEG_GZIP_ARCHIVE_NAME);
+    if let Err(error) = fs::remove_file(&gz_archive) {
+        error!(%error, "Failed to delete downloaded archive at {gz_archive:?}.");
+    }
+}
+
+fn prepare_dependencies(executable_dir: &Path) -> Result<()> {
+    download_ffmpeg(executable_dir).with_context(|| "libav download failed.")?;
+
+    // Archive contains only `libav` directory in the root. This directory contains all smelter
+    // libav dependencies.
+    unpack_ffmpeg(executable_dir).with_context(|| "Failed to unpack downloaded archive")?;
+
+    Ok(())
+}
+
+fn download_ffmpeg(executable_dir: &Path) -> Result<()> {
+    // TODO: Download in chunks
+    let response = get(ffmpeg_url())
+        .with_context(|| format!("Failed to download libav libraries. URL: {}", ffmpeg_url()))?;
+    let content = response.bytes()?;
+
+    fs::write(executable_dir.join(FFMPEG_GZIP_ARCHIVE_NAME), &content)
+        .with_context(|| "Failed to save downloaded libav libraries to file")?;
+    Ok(())
+}
+
+fn unpack_ffmpeg(executable_dir: &Path) -> Result<()> {
+    let gz_archive_path = executable_dir.join(FFMPEG_GZIP_ARCHIVE_NAME);
+
+    let gz_input = BufReader::new(
+        File::open(&gz_archive_path).with_context(|| "Failed to open downloaded archive")?,
+    );
+    let mut tar_bytes: Vec<u8> = vec![];
+
+    let mut decoder = flate2::bufread::GzDecoder::new(gz_input);
+
+    let decompressed_bytes = std::io::copy(&mut decoder, &mut tar_bytes)
```

Comment:
```suggestion
    let mut tar_bytes: Vec<u8> = vec![];
    let decompressed_bytes = std::io::copy(&mut decoder, &mut tar_bytes)
```
also you are naming `_bytes` variables with actual data and variables that store the number of bytes, maybe use `_size` or `_size_bytes`